### ISSUE TITLE
Add habitat to homebrew

### ISSUE THREAD

**Issue by bodymindarts** [habitat-sh/habitat#1896]
I tried adding habitat to the S&W brew tap but have difficulties finding a link that will deterministically download a specific version and is easily discoverable.

**Comment by smith** [comment]
It would not qualify as "easily discoverable", but https://github.com/habitat-sh/habitat/blob/master/components/hab/install.sh takes these options:

```
FLAGS:
    -c    Specifies a channel [values: stable, unstable] [default: stable]
    -h    Prints help information
    -v    Specifies a version (ex: 0.15.0/20161222215311)
```

**Comment by bodymindarts** [comment]
interesting. That script can perform install of a specific version but discovering the `0.15.0/20161222215311` part is the problem. How do I can I programatically get the timestamp of `0.17.0` for example.

**Comment by smith** [comment]
I think that what would need to happen here are improvements to the installer script.

If you type `install.sh -v 0.17.0` you get the exact same result if you type `install.sh -v 🐣` or any version that does not match an exact version/release that exists on Bintray.

The two things that need to be fixed that may get this working for you are:

* Give a failure message and exit non-zero if the version you are asking for is not found
* Allow specifying only a version without a release and make the program get the latest release for that version

**Comment by eeyun** [comment]
I agree here. I think this could be chocked up to a change in the "expected behavior" realm. 

I'd take this a step further and say we ought to get the install.sh available as a  first-class citizen with instructions on how to use it via the website.

**Comment by fnichol** [comment]
@bodymindarts All Linux `core/hab` [releases](https://app.habitat.sh/#/pkgs/core/hab) after June 2016 (when we launched) will have a corresponding Bintray version. Unfortunately our Depot currently doesn't have support for multiple platform/target packages (i.e. linux/64bit vs mac/64bit). We are working towards this however, and once we've landed the requisite work, you'll be able to use a Depot API endpoint for this query for all the Mac and Windows packages too. Once the multi-target Depot support is working we should be able to work towards eliminating Bintray as the initial download point and fold that into the Depot itself.

The reason that the [`install.sh` script](https://github.com/habitat-sh/habitat/blob/master/components/hab/install.sh) only takes a `VERION/RELEASE` "version" string is that each Bintray version is more like a compound key of a Habitat version string and release string joined with a dash (this was the closest we could get at the time to fit the notion of Habitat versioning into the Bintray API).

So, why all the background? That's essentially why it looks like we have random Mac and Windows releases with `VERSION-RELEASE` strings and a lacking API to find the latest. I would like us shortly to be able to use the Depot itself as that API endpoint, regardless of whether you want Linux, Mac, Windows, or future platform targets.

At the moment, adding more version string parsing logic into that `install.sh` script is going to make things more complicated as we need those inputs to pass both Bintray and possibly Depot API endpoints. Perhaps for the time being, we could detect-and-fail if a user provides a value to the `-v` option, but it isn't of the simplified egrep-form `'.+/[0-9]{14}$'` (i.e. make sure the string ends with a forward slash, followed by a Habitat release timestamp)?
